exasol_integration_test_docker_environment/lib/docker/images/create/utils/file_directory_list_hasher.py

```diff
@@ -94,16 +112,22 @@ def hash(self, files_and_directories: List[PathMapping]) -> bytes:
         hashes = self.compute_hashes(directory_mapping_result)
         return self._reduce_hash(hashes)
 
-    def validate(self, mappings: List[DestinationMapping]) -> None:
-        # Verify that there are no duplicate mappings to same destination
-        destinations = [p.destination for p in mappings]
-        if len(set(destinations)) != len(destinations):
-            raise AssertionError(f"Directory content for hashing contains duplicates: {destinations}")
+    def check_no_duplicate_destinations(self, mappings: List[DestinationMapping]) -> None:
+        """
+        Verify that there are no duplicate mappings to same destination.
+        This can happen, if the destination of two mappings are equal and the two sources contains the same sub-path
+        structure; or if the destination of two mappings is the same file.
+        """
+        destination_paths = [p.destination_path for p in mappings]
```

Comment:
needs to construct a set

Reply:
as discussed on phone, this works correctly as is. But to keep the code better understandable I changed it.